frontend/src/components/ImplementationReportView/DialectCompliance.tsx

```diff
@@ -42,7 +42,18 @@ const DialectCompliance: React.FC<{
               .map(([dialectName, result], index) => {
                 return (
                   <tr key={index}>
-                    <td>{Dialect.forPath(dialectName).uri}</td>
+                    <td>
+                      {Dialect.forPath(dialectName).uri}{" "}
+                      <img
+                        width={"auto"}
+                        height={20}
+                        src={`https://img.shields.io/endpoint?url=https%3A%2F%2Fbowtie-json-schema.github.io%2Fbowtie%2Fbadges%2F${encodeURIComponent(
```

Comment:
I think logic for constructing this image source can be extracted to a function. Also maybe it would be more clear if you actually URL encoded whole query param - so first construct something like this *https://bowtie-json-schema.github.io/bowtie/badges/java-dev.harrel.json-schema/supported_versions.json* and then URL encode whole thing

Reply:
Also the URL isn't the right one -- it should be https://bowtie.report -- but really, please use whatever way I presume vite/react has for getting the current base URI, which is even better, and will work even in the rare event we ever change domains.

Reply:
Hmm, you're right - in `index.tsx` we compute the URI for report fetching:
```ts
const reportHost =
  import.meta.env.MODE === "development"
    ? "https://bowtie.report"
    : window.location.href;
const reportUri = new URI(reportHost).directory(import.meta.env.BASE_URL);
```
I guess this could be extracted to a separate file and exported, so it could be used anywhere in code. 

I believe this URI will work fine also for badges but I haven't given much thought to this so maybe it would need some modifications

Reply:
> I think logic for constructing this image source can be extracted to a function. Also maybe it would be more clear if you actually URL encoded whole query param - so first construct something like this _https://bowtie-json-schema.github.io/bowtie/badges/java-dev.harrel.json-schema/supported_versions.json_ and then URL encode whole thing
I used encodedURIComponent for implementation.language to encode "C++" string to URL encoded format `C%2B%2B`.
Also encoding the whole string is isn't helpful. 
![image](https://github.com/bowtie-json-schema/bowtie/assets/43758514/2ccf9839-022b-4d99-a802-02ecee20f496)

Reply:
> Also encoding the whole string is not helpful.

What do you mean by that?

Reply:
I implemented above logic by encoding the query parameter URL.  To make this logic modular where should we create the file to store the required function? Like the file path in the folder structure, something like libs or utils folder maybe.

```jsx                      
<img
    alt={`${Dialect.forPath(dialectName).badgeName}`}
    src={`https://img.shields.io/endpoint?url=${encodeURIComponent(
      `https://bowtie-json-schema.github.io/bowtie/badges/${
        implementation.language
      }-${implementation.name}/compliance/${
        Dialect.forPath(dialectName).badgeName
      }.json`
    )}`}
  />`